src/utils/sources.ts

```diff
@@ -0,0 +1,17 @@
+import type { Source } from '@pdc/sdk';
+
+/**
+ * Iterates through a list of Sources to find the Source with the specified
+ * data provider shortcode
+ * Returns a Source with the specified data provider shortcode,
+ * or undefined if no such source exists
+ * @param {Source[]} entries an array of Sources
+ * @param {string} systemSourceShortCode the shortcode for the system source
+ * @return {Source | undefined} the system source, or undefined if it does not exist
+ */
+const getSystemSource = (entries: Source[], systemSourceShortcode: string) =>
```

Comment:
Unfortunate that this needs to be done but a clean and concise implementation given that it needs to be done!